### PR TITLE
fix: [CDS-46047]: type support for single env fields

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

The following changes have been made as part of this ticket
1. Removed execution time input support env group, environment, infra & clusters. All other fields have execution time support
2. Added expression support for all fields except env group, environment, infra & clusters. Expression support for single env, single infra in runtime input form is taken care of by another ticket - [PIE-6130: Environments V2: Environments and Infrastructure Definition expression support in run pipeline modalIN PROGRESS](https://harness.atlassian.net/browse/PIE-6130)

#### Screenshots

Now:

https://user-images.githubusercontent.com/96711906/202531372-c8f2f7ba-397c-4580-9dca-a09165639e8c.mov

Before:

https://user-images.githubusercontent.com/96711906/202531386-3e22ac19-ff52-4a3e-a381-35e970ad60b2.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 1]
```diff
@@ -1042,11 +1042,9 @@ export function StageInputSetFormInternal({
         deploymentStageTemplate={deploymentStageTemplate}
         allowableTypes={
           scope === Scope.PROJECT
-            ? ((allowableTypes as MultiTypeInputType[])?.filter(
-                item => item !== MultiTypeInputType.EXPRESSION
-              ) as AllowedTypes)
-            : ((allowableTypes as MultiTypeInputType[])?.filter(
-                item => item !== MultiTypeInputType.FIXED && item !== MultiTypeInputType.EXPRESSION
+            ? allowableTypes
+            : ((allowableTypes as MultiTypeInputType[]).filter(
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:1046]:
```suggestion
            : ((allowableTypes as MultiTypeInputType[])?.filter(
```

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-cd

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-pipeline

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-rest

**Comment by AshwinBhatkal-Harness** [comment]
trigger rebase

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**